pkg/daemon/update.go

```diff
@@ -788,6 +788,25 @@ func (dn *Daemon) deleteStaleData(oldConfig, newConfig *mcfgv1.MachineConfig) er
 				}
 				glog.V(2).Infof("Removing file %q completely", f.Path)
 			} else if _, err := os.Stat(origFileName(f.Path)); err == nil {
+				// Add a check for backwards compatibility: basically if the file doesn't exist in /usr/etc
+				// and no rpm is claming it, we assume that the orig file came from a wrongful backup of a
+				// MachineConfig file instead of a RHCOS file. See https://bugzilla.redhat.com/show_bug.cgi?id=1814397
+				if strings.HasPrefix(origFileName(f.Path), "/etc") {
+					if _, err := os.Stat("/usr" + origFileName(f.Path)); err != nil {
```

Comment:
should this be just `f.Path`? `/usr/etc` contains paths _in_ `/etc` whether `origFileName` turns `f.Path` into `/etc/machine-config-daemon/orig/f.Name` which won't live by default under `/usr/etc` which stores only RHCOS default